src/hotspot/share/utilities/powerOfTwo.hpp

```diff
@@ -121,10 +121,17 @@ inline T next_power_of_2(T value)  {
 }
 
 // Find log2 value greater than this input
-template <typename T, ENABLE_IF(std::is_integral<T>::value)>
-inline T ceil_log2(T value) {
-  T ret;
-  for (ret = 1; ((T)1 << ret) < value; ++ret);
+template <typename T, typename U = typename std::make_unsigned<T>::type, ENABLE_IF(std::is_integral<T>::value)>
+inline int ceil_log2(T value) {
+  assert(value > 0, "Invalid value");
+  U unsigned_value = static_cast<U>(value);
+  int max_bits = sizeof(U) * 8;
+  int ret;
+  for (ret = 0; ret < max_bits; ++ret) {
+    if ((U(1) << ret) >= unsigned_value) {
+      break;
+    }
+  }
```

Comment:
But I think this could all be much "simpler"?  I think this works:
```
template<typename T, ENABLE_IF(std::is_integral<T>::value)>
int ceil_log2(T value) {
  assert(value > T(0), "Invalid value");
  return log2i_graceful(value - 1) + 1;
}
```

Reply:
Sure, I updated to your suggestion instead.

Reply:
The use of `T(0)` in the assert isn't necessary; just use `0`.  Sorry about that.

Note that I didn't actually test that suggestion, just pencil and paper hand checked it.  It looks like the new tests
cover the needed values though. Except shouldn't there be a death test for 0?  Just one, I don't think you need
to death-test all covered types; death tests are kind of expensive to execute, since they need their own VM.

Reply:
Regarding death tests - look for tests using of `TEST_VM_ASSERT` and `TEST_VM_ASSERT_MSG`.
They are only available in debug builds, and uses are wrapped in `#ifdef ASSERT`.